src/community/json-ld/src/main/java/org/geoserver/jsonld/expressions/XPathFunction.java

```diff
@@ -0,0 +1,57 @@
+/* (c) 2019 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.jsonld.expressions;
+
+import static org.geotools.filter.capability.FunctionNameImpl.parameter;
+
+import org.geotools.feature.GeometryAttributeImpl;
+import org.geotools.filter.AttributeExpressionImpl;
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.locationtech.jts.geom.Geometry;
+import org.opengis.filter.capability.FunctionName;
+import org.opengis.filter.expression.Expression;
+import org.xml.sax.helpers.NamespaceSupport;
+
+public class XPathFunction extends FunctionExpressionImpl {
+
+    public static FunctionName NAME =
+            new FunctionNameImpl(
```

Comment:
You should be able to provide a description of the function here also?

aside: This function is generally useful and may be appropriate to contribute to geotools for wider use :)

Reply:
fixed